js/ui/dash.js

```diff
@@ -343,7 +343,8 @@ const DashActor = new Lang.Class({
         let layout = new Clutter.BoxLayout({ orientation: Clutter.Orientation.VERTICAL });
         this.parent({ name: 'dash',
                       layout_manager: layout,
-                      clip_to_allocation: true });
+                      clip_to_allocation: true,
+                      visible: false });
```

Comment:
Nice! I believe that since the actor is never created now, you should not need to make the dash invisible, or even need to remove the redisplay code below though.